cli/src/lib/init/stoatActionHelpers.ts

```diff
@@ -0,0 +1,127 @@
+import { getLocationForJsonPath, parseWithPointers } from '@stoplight/yaml';
+import chalk from 'chalk';
+import fs from 'fs';
+import inquirer from 'inquirer';
+import * as yaml from 'js-yaml';
+import path from 'path';
+
+import { getGitRoot } from '../pathHelpers';
+import { addStoatActionToYaml } from './configFileHelpers';
+
+export interface GhJob {
+  name: string;
+  workflowFile: string;
+}
+
+function addStoatAction(jobs: GhJob[]) {
+  jobs.forEach((job) => {
+    const newYaml = addStoatActionToYaml(job);
+    fs.writeFileSync(job.workflowFile, newYaml);
+    console.log(`Added Stoat action for job: ${job.workflowFile} (${job.name})`);
+  });
+}
+
+async function approveWorkflows(jobs: GhJob[]) {
+  for (const job of jobs) {
+    const questions = [
+      {
+        type: 'confirm',
+        name: 'confirm',
+        message: `Add Stoat action for: ${job.workflowFile} (${job.name})`,
+        default: false
+      }
+    ];
+
+    await inquirer.prompt(questions).then((answers: any) => {
+      if (answers.confirm) {
+        addStoatAction([job]);
+      } else {
+        console.log(`Skipping adding Stoat action for job: ${job.workflowFile} (${job.name})`);
+      }
+    });
+  }
+
+  console.log('\n');
+}
+
+function getRelevantJobs(dir: string): GhJob[] {
+  const filenames = fs.readdirSync(dir);
+  const matchingJobs: GhJob[] = [];
+
+  for (const filename of filenames) {
+    const fullPath = path.join(dir, filename);
+    const isDirectory = fs.lstatSync(fullPath).isDirectory();
+
+    if (isDirectory) {
+      matchingJobs.push(...getRelevantJobs(fullPath));
+    } else if (path.extname(filename) === '.yml' || path.extname(filename) === '.yaml') {
+      const contents = fs.readFileSync(fullPath, 'utf8');
+
+      try {
+        const workflow: any = yaml.load(contents);
+
+        if (('jobs' in workflow && 'on' in workflow && 'pull_request' in workflow.on) || 'push' in workflow.on) {
+          for (const job in workflow.jobs) {
+            if ('steps' in workflow.jobs[job]) {
+              const steps = workflow.jobs[job].steps as any[];
+
+              if (
+                steps.length > 0 &&
+                steps.filter((step) => 'uses' in step && step.uses.includes('stoat-dev/stoat-action')).length === 0
```

Comment:
Nitpick. The `Array#some` method tests whether there is at least one element that satisfies the condition. Using this method eliminates the necessity to manually check the length.

```suggestion
                !steps.some((step) => 'uses' in step && step.uses.includes('stoat-dev/stoat-action'))
```